src/main/java/com/thealgorithms/maths/Means.java

```diff
@@ -0,0 +1,57 @@
+package com.thealgorithms.maths;
+
+import java.util.stream.StreamSupport;
+import org.apache.commons.collections4.IterableUtils;
```

Comment:
I am not sure what is the policy regarding introducing new _external_ dependencies. @siriak can you comment on that?

Reply:
If iterable is a collector then time complexity is O(1). And if its not a collection then worst case O(n).

Reply:
Thanks. You are [right](https://commons.apache.org/proper/commons-collections/apidocs/src-html/org/apache/commons/collections4/IterableUtils.html#line.806) regarding the time complexity.

Reply:
We don't add new dependencies very often, but I think it's a reasonable one to have.